### PR TITLE
Remove Firecloud from Study Server

### PR DESCRIPTION
**Pull request by MocanaAtBroad**
## Context
Note that we are branching off study-server-parallel-test . Part of the changes here have to do with stopping the generation of config files that still refer to Firecloud, the use of the command line that includes the FireCloud options etc.
Not using it so tried to remove every reference I could  of FireCloud.
There is still a bit in left in User DAO that IMO seemed more trouble than was worth to remove.

If you want to run the tests, you can execute:
```
pepper-apis/scripts/run_ci_tests.sh remove_firecloud
```

## Checklist

- [ X] I have labeled the type of changes involved using the `C-*` labels.
- [X ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [ X] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing
We already have tests that have been modified to account for the removal of functionality.
- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [X ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -482,6 +481,7 @@ jobs:
       - run-maven-with-pepper-options:
           phase: test
           skip_tests: false
+          skip_checkstyle: false
           additional_options: '-Dtest=ServerInSameProcessIntegrationTestSuite'
           restore_m2_cache: true
```
Comment (MocanaAtBroad) [inline .circleci/config.yml]: This leaked from parallel-test branch. Pay no attention.

File: pepper-apis/src/test/java/org/broadinstitute/ddp/route/AuthFilterRouteTest.java [hunk 2]
```diff
@@ -342,7 +352,7 @@ public void testGovernedStudyParticipants_failsWhenRequestedUserIsNotOperatorIts
 
     @Test
     public void testAdminStudiesRequestFailsWithNoToken() throws Exception {
-        assert401NoBodyForStandardVerbs(buildAdminStudiesUrl(), null);
+        assert404ForStandardVerbs(buildAdminStudiesUrl(), null);
```
Comment (yufengwng) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/route/AuthFilterRouteTest.java]: Why not just remove the tests altogether?

File: pepper-apis/config/housekeeping_startup.sh.ctmpl [hunk 3]
```diff
@@ -5,7 +5,6 @@ echo 'Starting housekeeping...'
 {{with $version := env "VERSION"}}
 {{with $conf := vault (printf "secret/pepper/%s/%s/conf" $environment $version)}}
 
-CORE_COMMAND="-Dddp.firecloudKeysDir=fc_keys -Dconfig.file=config/application.conf -Xms256m -Xmx512m -jar target/Housekeeping.jar"
```
Comment (yufengwng) [inline pepper-apis/config/housekeeping_startup.sh.ctmpl:8]: Looks like we got a little too eager with the deletion here?

File: pepper-apis/src/main/java/org/broadinstitute/ddp/constants/RouteConstants.java [hunk 4]
```diff
@@ -88,7 +88,6 @@
         public static final String USER_STUDY_EXIT = String.format(
                 BASE + "/user/%s/studies/%s/exit", PathParam.USER_GUID, PathParam.STUDY_GUID);
 
-        public static final String ADMIN_BASE = BASE + "/admin";
         public static final String ADMIN_STUDIES = BASE + "/admin/studies";
         public static final String ADMIN_WORKSPACES = BASE + "/admin/workspaces";
         public static final String EXPORT_STUDY = fmt(BASE + "/admin/studies/%s/export", PathParam.STUDY_GUID);
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/constants/RouteConstants.java]: These can also go?

**Review by yufengwng — COMMENTED** [review]
Looks good. A few nitpicks:

* The `study-server-parallel-tests` PR is already HUGE. I recommend splitting these changes off into it's own separate PR.
* I don't think this is "breaking change"? We don't have any production participant/study-staff facing features that depend on this code. And none of the APIs are documented in our OpenAPI specs.
* I suggest pinging @SimoneMaiwald or @pegahtah. I recall we did some experiment to integrate some FireCloud stuff into DSM frontend angular code but not sure if any of that stuff got merged.

**Comment by SimoneMaiwald** [comment]
Firecloud stuff never got merged into dsm develop branch. Can be removed from ddp-study-manager point of view

**Comment by MocanaAtBroad** [comment]
> Looks good. A few nitpicks:
> 
> * The `study-server-parallel-tests` PR is already HUGE. I recommend splitting these changes off into it's own separate PR.

Not sure what you mean. You mean branch off `develop` instead? Needed to change stuff that currently lives in `parallel` so that is why I branched off that.

> * I don't think this is "breaking change"? We don't have any production participant/study-staff facing features that depend on this code. And none of the APIs are documented in our OpenAPI specs.

I thought the labeled indicated that "might break" stuff (the text label is truncated so can't read in full). Was not totally sure that something else might not break so selected that. But if you are certain that nothing breaks than we can remove it (does it matter at this point?)

> * I suggest pinging @SimoneMaiwald or @pegahtah. I recall we did some experiment to integrate some FireCloud stuff into DSM frontend angular code but not sure if any of that stuff got merged.

Looks like they have chimed in!

File: pepper-apis/config/housekeeping_startup.sh.ctmpl (hunk 3 above)
Comment (MocanaAtBroad) [reply pepper-apis/config/housekeeping_startup.sh.ctmpl:8]: Got carried away! 
fd8080d

**Comment by MocanaAtBroad** [comment]
Thanks for feedback:
413bb03
fd8080d
3f29dc0

File: .circleci/config.yml [hunk 5]
```diff
@@ -275,7 +274,7 @@ commands:
               echo "Deleting version $CURRENT_VERSION of service $SERVICE"
               gcloud app versions delete --quiet --service "$SERVICE" "$CURRENT_VERSION" --project "broad-ddp-${ENVIRONMENT}"
             done;
-            : # Need to make sure that exit code for step is a 0 even if versions delete failed
+            exit 0
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:278]: Not sure why this showing up here. This change already in 'parallel' branch.

File: pepper-apis/api-build.sh [hunk 6]
```diff
@@ -53,16 +53,17 @@ function push_docs_image {
...
-  if [[ "$1" == "local" ]]; then
+  if [[ "$1" == "local" ]] || [[ "$1" == "gae" ]]; then
       echo "rendering runtime configs locally"
-      INPUT_DIR=config NO_SYSLOG=true DOCS_PROXIED_HOST=$DOCS_PROXIED_HOST NGINX_PROXIED_HOST=$NGINX_PROXIED_HOST OUTPUT_DIR=$OUTDIR ENVIRONMENT=$ENVIRONMENT DIR=$DIR/$OUTDIR VERSION=$VERSION ruby configure.rb -y
+      INPUT_DIR=config NO_SYSLOG=true DOCS_PROXIED_HOST=$DOCS_PROXIED_HOST NGINX_PROXIED_HOST=$NGINX_PROXIED_HOST OUTPUT_DIR=$OUTDIR ENVIRONMENT=$ENVIRONMENT DIR=$DIR/$OUTDIR VERSION=$VERSION GAE=$GAE ruby configure.rb -y
       cp $OUTDIR/docker-compose.yaml $DIR
-  elif [[ "$1" == "gae" ]]; then
-      echo "gae configs"
-      INPUT_DIR=config NO_SYSLOG=true DOCS_PROXIED_HOST=$DOCS_PROXIED_HOST NGINX_PROXIED_HOST=$NGINX_PROXIED_HOST OUTPUT_DIR=$OUTDIR ENVIRONMENT=$ENVIRONMENT DIR=$DIR/$OUTDIR VERSION=$VERSION GAE=true ruby configure.rb -y
   fi
```
Comment (MocanaAtBroad) [inline pepper-apis/api-build.sh:67]: Also from 'parallel'

File: pepper-apis/config/manifest.rb [hunk 7]
```diff
@@ -10,12 +10,5 @@
...
 render "application.conf.ctmpl"
-render "testing-circleci.conf.ctmpl"
 copy_file "post_deploy_smoketest.sh"
 copy_file "waf.conf"
-fc_keys = './fc_keys'
-Dir.mkdir(fc_keys)
-# at some point managing fc keys this way will become a headache...
-# service account keys below here are the ones that will be used by the backend and should have file names
-# that are reflected in the database (as just file names, not paths)
-copy_secret_from_path "secret/pepper/#{$env}/#{$version}/test-firecloud-service-account.json",nil,fc_keys + "/ddp-testing.json"
```
Comment (MocanaAtBroad) [inline pepper-apis/config/manifest.rb:21]: This is from parallel and this branch.

File: pepper-apis/config/testing-circleci.conf.ctmpl [hunk 8]
```diff
@@ -1,182 +0,0 @@
-# Use this file to run local tests using a disposable in-memory database
-# while connecting to other environment-specific resources
-
```
Comment (MocanaAtBroad) [inline pepper-apis/config/testing-circleci.conf.ctmpl:3]: Removed in 'parallel'

File: pepper-apis/config/testing-inmemorydb.conf.ctmpl [hunk 9]
```diff
@@ -61,27 +68,9 @@
     "easyPostApiKey": "{{$testingConf.Data.easyPostApiKey}}",
     # by default, boot in-processs for testing.  change rendered .conf file manually for local development
     "bootTestAppInSeparateProcess": false,
-    "baseTestUrl": "http://localhost:5555",
+    "baseTestUrl": "http://localhost:5556",
     "socialTestEmail": "{{$testingConf.Data.socialTestEmail}}",
     "socialTestPassword": "{{$testingConf.Data.socialTestPassword}}",
```
Comment (MocanaAtBroad) [inline pepper-apis/config/testing-inmemorydb.conf.ctmpl:73]: This changes from 'parallel' too.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/PubSubEmulator.java [hunk 10]
```diff
@@ -168,7 +168,7 @@ private static boolean isExternalEmulatorRunning() {
                 CloseableHttpClient httpClient = HttpClients.createDefault();
         ) {
             CloseableHttpResponse response =
-                    httpClient.execute(new HttpGet("http://" + EMULATOR_HOST));
+                    httpClient.execute(new HttpGet(PUBSUB_BASE_URL));
             return response.getStatusLine().getStatusCode() == 200;
```
Comment (MocanaAtBroad) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/util/PubSubEmulator.java:172]: From 'parallel'

File: pepper-apis/src/test/java/org/broadinstitute/ddp/util/TestDataSetupUtil.java [hunk 11]
```diff
@@ -501,8 +500,6 @@ public static synchronized void insertStaticTestData() {
         scripts.add(BASELINE_SEED_TEST_DATA);
         scripts.add(MIGRATE_LEGACY_STUDY_CLIENT_TENANT_AND_ENABLE_CONSTRAINTS);
         try {
-            Driver driver = new com.mysql.cj.jdbc.Driver();
-
             for (String script : scripts) {
```
Comment (MocanaAtBroad) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/util/TestDataSetupUtil.java:503]: From 'parallel'

**Review by yufengwng — APPROVED** [review]
LGTM

File: .circleci/config.yml (hunk 5 above)
Comment (yufengwng) [reply .circleci/config.yml:278]: It might have been cleaner to split this off to a separate PR branched off of `develop` :)

**Pull request merged**